internal/inventory/gcpfetcher/fetcher_assets.go

```diff
@@ -103,3 +108,74 @@ func (f *assetsInventory) fetch(ctx context.Context, assetChan chan<- inventory.
 		)
 	}
 }
+
+func (f *assetsInventory) findRelatedAssetIds(subType inventory.AssetSubType, item *gcpinventory.ExtendedGcpAsset) []string {
+	ids := []string{}
+	ids = append(ids, item.Ancestors...)
+	if item.Resource != nil {
+		ids = append(ids, item.Resource.Parent)
+	}
+
+	ids = append(ids, f.findRelatedAssetIdsForSubType(subType, item)...)
+
+	ids = lo.Compact(ids)
+	ids = lo.Uniq(ids)
+	return ids
+}
+
+func (f *assetsInventory) findRelatedAssetIdsForSubType(subType inventory.AssetSubType, item *gcpinventory.ExtendedGcpAsset) []string {
+	ids := []string{}
+
+	var fields map[string]*structpb.Value
+	if item.Resource != nil && item.Resource.Data != nil {
+		fields = item.GetResource().GetData().GetFields()
+	}
+
+	switch subType {
+	case inventory.SubTypeGcpInstance:
+		if v, ok := fields["networkInterfaces"]; ok {
+			for _, networkInterface := range v.GetListValue().GetValues() {
+				networkInterfaceFields := networkInterface.GetStructValue().GetFields()
+				ids = appendIfExists(ids, networkInterfaceFields, "network")
```

Comment:
Why do we need to have a map with the name of the field?

Reply:
I'm not entirely sure what you are asking about. Is it about "Why do we have to call `...GetFields(), which returns a map, that we have to query further?"?
If so, then the answer is Google's SDK is made that way 🤷